pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/ZKMetadataStore.java

```diff
@@ -386,7 +386,13 @@ private void internalStorePut(OpPut opPut) {
                                 put(opPut.getPath(), opPut.getData(), Optional.of(-1L)).thenAccept(
                                                 s -> future.complete(s))
                                         .exceptionally(ex -> {
-                                            future.completeExceptionally(MetadataStoreException.wrap(ex.getCause()));
+                                            if (ex instanceof BadVersionException) {
```

Comment:
BadVersion exception happened with concurrent write to same ZNODE.  I think it's ok when some writer failed with BadVersionException because of competing failed. @wangjialing218

Reply:
The code here is handling the put request without `expectedVersion`, that means the put operation should succeed in case of concurrent write. we should not throw BadVersionException here, which means the expected version doesn't match the actual version of the data.